cpp/test/Ice/optional/Test.ice

```diff
@@ -161,20 +161,20 @@ class OptionalWithCustom
 
 class E
 {
-    A ae;
+    FixedStruct fse;
 }
 
 class F extends E
 {
-    optional(1) A af;
+    optional(1) FixedStruct fsf;
 }
 
-class G1
+struct G1
```

Comment:
G1/G2 and G are apparently used for opG below. It's not clear to me what is special with this test, especially now that G1 and G2 are plain structs.

Reply:
All of these `G` types are for testing the combination of optionals and `marshaled-result`:
```
    ["marshaled-result"] G opMG1();
    ["marshaled-result"] G opMG2(G p1, out G p2);
```

We have a 4 marshaled-result tests in total.
The first uses _optional structs_ (which hold a single required byte),
the second uses an _optional sequence_, the third uses an _optional dictionary_.

But none of these test marshaled-result with _optional members_.
Hence, I kept the 4th test, since, that's what it tests. A class which holds a combination of optional and required data members. And this test doesn't seem class specific. All we need is a class which holds some optional things internally.

Reply:
Now, maybe we could remove `opG()`.
It seems we added it as part of a bug-fix: `Fixed ICE-6602: optionals marshaling bug`, so to play it safe I kept it.
I couldn't find any more detailed explanation in the logs, and forgot how to log into JIRA.